great_expectations/expectations/expectation.py

```diff
@@ -1634,16 +1704,18 @@ def _get_renderer_diagnostics(
     def _get_execution_engine_diagnostics(
         metric_diagnostics_list: List[ExpectationMetricDiagnostics],
         registered_metrics: dict,
-        execution_engine_names: List[str] = [
-            "PandasExecutionEngine",
-            "SqlAlchemyExecutionEngine",
-            "SparkDFExecutionEngine",
-        ],
+        execution_engine_names: Optional[List[str]] = None,
     ) -> ExpectationExecutionEngineDiagnostics:
         """Check to see which execution_engines are fully supported for this Expectation.
 
         In order for a given execution engine to count, *every* metric must have support on that execution engines.
         """
+        if execution_engine_names is None:
```

Comment:
```suggestion
        if not execution_engine_names:
```
What should the behavior be if an empty list is provided?
I general I think we are overusing explicitly `None` checks, when `if TRUTHY` checks are more appropriate.

Reply:
Completely agree. I also updated this in the method above. ✅